src/accounts/pages/DownloadApp.tsx

```diff
@@ -1,53 +1,378 @@
+
+  // Update CSS variable dynamically
+  useEffect(() => {
+    if (app) {
+      updateThemeColors(app.accentColor, app.backgroundColor);
+    }
+  }, [status, app]);
+
+  useEffect(() => {
+    // Prevent useEffect from running twice which causes a race condition of the
+    // async handleRequests() between which publicKey is sent in the request and this
+    // conflicts in the publicKey that is saved in localStorage
+    if (!rendered) {
+      rendered = true;
+    } else {
+      return;
+    }
+
+    onLoad();
+  }, []);
+
+  // check if user logged in and if not starts login process from URL parameters
+  async function onLoad() {
+    try {
+      const urlParams = new URLSearchParams(window.location.search);
+      const payload = urlParams.get("payload");
+
+      if (payload) {
+        setPayload(payload);
+      }
+      await checkLoggedIn();
+    } catch (e) {
+      if (
+        e instanceof SdkError &&
+        (e.code === SdkErrors.AccountNotFound ||
+          e.code === SdkErrors.UserNotLoggedIn ||
+          e.code === SdkErrors.AccountDoesntExist)
+      ) {
+        loginToTonomyAndSendRequests();
+      } else {
+        errorStore.setError({ error: e, expected: false });
+      }
+    }
+  }
+
+  // check if user is already logged in
+  async function checkLoggedIn() {
+    debug("checkLoggedIn()");
+
+    // only clear state if the accountsLogin is not set (i.e. they are not navigating from other part of the website)
+    const user = await ExternalUser.getUser({
+      autoLogout: accountsLogin ? false : true,
+    });
+
+    setStatus("connecting");
+
+    loginToTonomyAndSendRequests(user);
+  }
+
+  async function getAppDetails(loginRequest: LoginRequest) {
+    debug("getAppDetails()", loginRequest.getPayload().origin);
+
+    const app = await App.getApp(loginRequest.getPayload().origin);
+
+    setApp(app);
+  }
+  async function terminateLogin(error): Promise<string> {
+    const { requests } = await getLoginRequestFromUrl();
+    const managedRequests = new RequestsManager(requests);
+
+    const externalLoginRequest =
+      managedRequests.getLoginRequestWithDifferentOriginOrThrow();
+
+    const managedResponses = new ResponsesManager(managedRequests);
+
+    return (await terminateLoginRequest(managedResponses, "mobile", error, {
+      callbackOrigin: externalLoginRequest.getPayload().origin,
+      callbackPath: externalLoginRequest.getPayload().callbackPath,
+    })) as string;
+  }
+
+  const onRefresh = async () => {
+    try {
+      const callbackUrl = await terminateLogin({
+        code: SdkErrors.UserRefreshed,
+        reason: "User refreshed during login",
+      });
+
+      window.location.href = callbackUrl;
+    } catch (e) {
+      errorStore.setError({ error: e, expected: false });
+    }
+  };
+
+  // creates SSO login request and sends the login request to Tonomy ID, via URL or communication server
+  async function loginToTonomyAndSendRequests(user?: ExternalUser) {
```

Comment:
a lot of functions here that you have copied from Login.tsx. Also some in Home.tsx. Please make a shared file that has all the shared functions in it.